test/test_connectionpool.py

```diff
@@ -533,6 +535,21 @@ def _test(exception):
         _test(SocketError)
         _test(ProtocolError)
 
+    def test_raise_timeout_error(self):
```

Comment:
So it turns out the branch you're adding this for was [added 7 years ago](https://github.com/urllib3/urllib3/pull/493) and has been unchanged since, tbh I'm not even sure how to trigger it since we never knew it wasn't covered. The test case supposedly triggering it is `with_dummserver/test_socketlevel.py:test_https_connection_read_timeout`

Reply:
i've moved this test to `with_dummserver/test_socketlevel.py:test_https_connection_read_timeout` and added a comment to make clear that we test both errors here.